Registry read: panic runtime error on unquoted reference

**Describe the bug**
When you pass in an unquoted registry path like SYSTEM\CurrentControlSet\Control\Lsa\LsaPid, a crash happens

**To Reproduce**
Just run the registry read command with an unquoted path:
```
registry read --hive HKLM SYSTEM\CurrentControlSet\Control\Lsa\LsaPid

panic: runtime error: slice bounds out of range [:-1]

goroutine 1 [running]:
github.com/bishopfox/sliver/client/command/registry.RegReadCmd(0xc00078e160, 0xc000564ae0)
        github.com/bishopfox/sliver/client/command/registry/reg-read.go:101 +0x6fe
github.com/bishopfox/sliver/client/command.BindCommands.func177(0xc00078e160, 0xc00078e160, 0x1)
        github.com/bishopfox/sliver/client/command/commands.go:1637 +0x5d
github.com/desertbit/grumble.(*App).RunCommand(0xc00068a0e0, 0xc0004386c0, 0x1, 0x4, 0x5, 0x8)
        github.com/desertbit/grumble@v1.1.1/app.go:269 +0x248
github.com/desertbit/grumble.(*App).runShell(0xc00068a0e0, 0xc0003ad538, 0x1)
        github.com/desertbit/grumble@v1.1.1/app.go:461 +0x5a5
github.com/desertbit/grumble.(*App).Run(0xc00068a0e0, 0x0, 0x0)
        github.com/desertbit/grumble@v1.1.1/app.go:393 +0x55e
github.com/bishopfox/sliver/client/console.Start(0x1411b48, 0xc00056c120, 0x1034558, 0x10340f8, 0x0, 0x0, 0x2)
        github.com/bishopfox/sliver/client/console/console.go:129 +0x3dd
github.com/bishopfox/sliver/client/cli.StartClientConsole(0x0, 0x0)
        github.com/bishopfox/sliver/client/cli/cli.go:100 +0x250
github.com/bishopfox/sliver/client/cli.glob..func1(0x18c0800, 0x1906db0, 0x0, 0x0)
        github.com/bishopfox/sliver/client/cli/cli.go:74 +0x7a
github.com/spf13/cobra.(*Command).execute(0x18c0800, 0xc000030210, 0x0, 0x0, 0x18c0800, 0xc000030210)
        github.com/spf13/cobra@v1.1.1/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x18c0800, 0x18d8b40, 0x0, 0xc000052778)
        github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.1.1/command.go:895
github.com/bishopfox/sliver/client/cli.Execute()
        github.com/bishopfox/sliver/client/cli/cli.go:105 +0x31
main.main()
        github.com/bishopfox/sliver/client/main.go:26 +0x25
```
**Expected behavior**
I expect it to work or yell at me and tell me to quote my path

**Screenshots**
N/A, see above crash dump

**Desktop (please complete the following information):**
Victim Machine:
OS: windows
Version: 10 build 19042 x86_64
Arch: amd64

**Additional context**
Sliver version information:
```
[*] Client v1.4.20 - 7a38ddffb60d9205b1cef1bac430af742eff4e99 - linux/amd64
    Compiled at 2021-07-23 12:35:49 -0400 EDT
    Compiled with go version go1.16.6 linux/amd64

[*] Server v1.4.20 - 7a38ddffb60d9205b1cef1bac430af742eff4e99 - linux/amd64
    Compiled at 2021-07-23 12:35:48 -0400 EDT
```

--- a/client/command/commands.go
+++ b/client/command/commands.go
@@ -1643,7 +1643,7 @@ func BindCommands(con *console.SliverConsoleClient) {
 		},
 		Flags: func(f *grumble.Flags) {
 			f.Int("t", "timeout", defaultTimeout, "command timeout in seconds")
-			f.String("H", "hive", "HKCU", "egistry hive")
+			f.String("H", "hive", "HKCU", "registry hive")
 			f.String("o", "hostname", "", "remote host to read values from")
 		},
 		HelpGroup: consts.SliverWinHelpGroup,

--- a/client/command/registry/reg-create.go
+++ b/client/command/registry/reg-create.go
@@ -49,12 +49,20 @@ func RegCreateKeyCmd(ctx *grumble.Context, con *console.SliverConsoleClient) {
 	if strings.Contains(regPath, "/") {
 		regPath = strings.ReplaceAll(regPath, "/", "\\")
 	}
-	slashIndex := strings.LastIndex(regPath, "\\")
-	key := regPath[slashIndex+1:]
-	regPath = regPath[:slashIndex]
+	pathBaseIdx := strings.LastIndex(regPath, `\`)
+	if pathBaseIdx < 0 {
+		con.PrintErrorf("invalid path: %s", regPath)
+		return
+	}
+	if len(regPath) < pathBaseIdx+1 {
+		con.PrintErrorf("invalid path: %s", regPath)
+		return
+	}
+	finalPath := regPath[:pathBaseIdx]
+	key := regPath[pathBaseIdx+1:]
 	createKeyResp, err := con.Rpc.RegistryCreateKey(context.Background(), &sliverpb.RegistryCreateKeyReq{
 		Hive:     hive,
-		Path:     regPath,
+		Path:     finalPath,
 		Key:      key,
 		Hostname: hostname,
 		Request:  con.ActiveSession.Request(ctx),

--- a/client/command/registry/reg-read.go
+++ b/client/command/registry/reg-read.go
@@ -72,6 +72,10 @@ func getType(t string) (uint32, error) {
 
 // RegReadCmd - Read a windows registry key: registry read --hostname aa.bc.local --hive HKCU "software\google\chrome\blbeacon\version"
 func RegReadCmd(ctx *grumble.Context, con *console.SliverConsoleClient) {
+	var (
+		finalPath string
+		key       string
+	)
 	session := con.ActiveSession.GetInteractive()
 	if session == nil {
 		return
@@ -93,15 +97,24 @@ func RegReadCmd(ctx *grumble.Context, con *console.SliverConsoleClient) {
 		con.PrintErrorf("You must provide a path")
 		return
 	}
+
 	if strings.Contains(regPath, "/") {
 		regPath = strings.ReplaceAll(regPath, "/", "\\")
 	}
-	slashIndex := strings.LastIndex(regPath, "\\")
-	key := regPath[slashIndex+1:]
-	regPath = regPath[:slashIndex]
+	pathBaseIdx := strings.LastIndex(regPath, `\`)
+	if pathBaseIdx < 0 {
+		con.PrintErrorf("invalid path: %s", regPath)
+		return
+	}
+	if len(regPath) < pathBaseIdx+1 {
+		con.PrintErrorf("invalid path: %s", regPath)
+		return
+	}
+	finalPath = regPath[:pathBaseIdx]
+	key = regPath[pathBaseIdx+1:]
 	regRead, err := con.Rpc.RegistryRead(context.Background(), &sliverpb.RegistryReadReq{
 		Hive:     hive,
-		Path:     regPath,
+		Path:     finalPath,
 		Key:      key,
 		Hostname: hostname,
 		Request:  con.ActiveSession.Request(ctx),

--- a/client/command/registry/reg-write.go
+++ b/client/command/registry/reg-write.go
@@ -71,9 +71,17 @@ func RegWriteCmd(ctx *grumble.Context, con *console.SliverConsoleClient) {
 	if strings.Contains(regPath, "/") {
 		regPath = strings.ReplaceAll(regPath, "/", "\\")
 	}
-	slashIndex := strings.LastIndex(regPath, "\\")
-	key := regPath[slashIndex+1:]
-	regPath = regPath[:slashIndex]
+	pathBaseIdx := strings.LastIndex(regPath, `\`)
+	if pathBaseIdx < 0 {
+		con.PrintErrorf("invalid path: %s", regPath)
+		return
+	}
+	if len(regPath) < pathBaseIdx+1 {
+		con.PrintErrorf("invalid path: %s", regPath)
+		return
+	}
+	finalPath := regPath[:pathBaseIdx]
+	key := regPath[pathBaseIdx+1:]
 	switch valType {
 	case sliverpb.RegistryTypeBinary:
 		var (
@@ -118,7 +126,7 @@ func RegWriteCmd(ctx *grumble.Context, con *console.SliverConsoleClient) {
 		Request:     con.ActiveSession.Request(ctx),
 		Hostname:    hostname,
 		Hive:        hive,
-		Path:        regPath,
+		Path:        finalPath,
 		Type:        valType,
 		Key:         key,
 		StringValue: stringValue,

